### PR TITLE
ENG-18665: Rename task classes

### PR DESCRIPTION
**Pull request by n3world**
Rename some of the task classes and update the corresponding documentation and variable names
ActionSchedule -> IntervalGenerator
ActionDelay -> Interval
DelayedAction -> ScheduledAction

### CODE REVIEW

File: src/frontend/org/voltdb/task/ActionScheduler.java [hunk 1]
```diff
@@ -18,18 +18,18 @@
...
- * Interface which generates actions to be performed as well as calculating the delay until that should be performed
- * according to a schedule
+ * Interface which generates actions to be performed as well as calculating the interval that should elapse before the
+ * action is performed.
  */
 public interface ActionScheduler extends ActionGeneratorBase {
     /**
      * This method is invoked for the first action to be performed. All subsequent invocation will be of
-     * {@link DelayedAction#getCallback()}. After the delay has elapsed and the action has been performed.
+     * {@link ScheduledAction#getCallback()}. After the interval has elapsed and the action has been performed.
```
Comment (rdykiel) [inline src/frontend/org/voltdb/task/ActionScheduler.java]: Review the grammar of those sentences. The third sentence is not a complete sentence. I suggest (if I understand how this works):

"This method is invoked for the first action to be performed; after this method returns, all subsequent invocations occur after the interval has elapsed and those invocations will be of ...."

Also I don't understand "invocation will be of.." what does 'of' mean?

File: src/frontend/org/voltdb/task/DelayIntervalGenerator.java [hunk 2]
```diff
@@ -20,19 +20,19 @@
 import java.util.concurrent.TimeUnit;
 
 /**
- * {@link ActionSchedule} implementation which executes actions at fixed delay between execution times
+ * {@link IntervalGenerator} implementation which executes actions at fixed delay between execution times
```
Comment (rdykiel) [inline src/frontend/org/voltdb/task/DelayIntervalGenerator.java]: "...which executes actions separated by a fixed delay"?

File: src/frontend/org/voltdb/task/IntervalGenerator.java [hunk 3]
```diff
@@ -18,17 +18,17 @@
...
 /**
- * Interface which calculates the length of delay until the next action should be performed according to this schedule
+ * Interface which calculates the time interval which should elapse until the next action should be performed
  */
-public interface ActionSchedule extends Initializable {
+public interface IntervalGenerator extends Initializable {
     /**
-     * This method is invoked for the delay of the first action to be performed. All subsequent invocation will be of
-     * {@link ActionDelay#getCallback()}
+     * This method is invoked for the interval of the first action to be performed. All subsequent invocation will be of
```
Comment (rdykiel) [inline src/frontend/org/voltdb/task/IntervalGenerator.java]: "Returns the interval before the first action" ?

File: src/frontend/org/voltdb/task/ScheduledAction.java [hunk 4]
```diff
@@ -17,39 +17,176 @@
...
+    /**
+     * @return Optional status message provided with any action
+     */
+    public String getStatusMessage() {
+        return m_action.getStatusMessage();
     }
 
-    @Override
+    /**
+     * Set the optional status massage which will be reported in the statistics for a task and if this is an
```
Comment (rdykiel) [inline src/frontend/org/voltdb/task/ScheduledAction.java]: "massage"????

File: src/frontend/org/voltdb/task/ActionScheduler.java (hunk 1 above)
Comment (n3world) [reply src/frontend/org/voltdb/task/ActionScheduler.java]: Updated to
This method is invoked only once to obtain the first action and interval. The action will be performed after the provided time interval has elapsed. After the action has been performed the {@link ScheduledAction#getCallback() callback} will be invoked with the {@link ActionResult result} of the action. The return of the callback will be used as the next action and interval.

**Pull request merged**